### DISCUSSION TITLE
Android keyboard backspace buffer

### DISCUSSION THREAD

**Discussion by sravyak** [sourcelair/xterm.js#4365] [Q&A]
I am using ng-terminal in my project. I observed this unexpected behaviour when I use terminal in android chrome. When I backspace in the terminal, I erase previous command input by 1 character and keep going to previous command history.

Browser and version: Android Chrome - 109.0.5414.46 xterm.js version: 5.0.0

The key issue I think we're facing, aside from the behaviour being odd, is the backspace buffer. I suspect that the keyboard has its own buffer of text that isn't being cleared when the app decides to consume the text, and when you backspace, it moves back in to this buffer This is why each word reappears when you backspace.

**Comment by Tyriar** [comment]
Issues like this around Android input are tracked in https://github.com/xtermjs/xterm.js/issues/3600. You're pretty much right where the textarea's text representation is difficult to what's on screen, so Android's IME doesn't play nicely.

**Reply by sravyak** [reply]
Thanks for the reply. Is it possible that there is any configuration setting that we can change to switch this behaviour or disabling the keyboard buffer and sending each key to the terminal on every key event.

**Reply by Tyriar** [reply]
I've been waiting for the community to step up and fix this one as it's a minor use can for vscode.dev. Detecting that it's Android and disabling some stuff seems reasonable
